### PR TITLE
refactor: make dcast.isVoidArrayLiteral private + move rawTypeMerge to dtemplate.d

### PR DESCRIPTION
**Pull request by RazvanN7**
...and make it private

### CODE REVIEW

File: src/ddmd/dcast.d [hunk 1]
```diff
@@ -2666,28 +2666,6 @@ extern (C++) bool isVoidArrayLiteral(Expression e, Type other)
     return (e.op == TOKarrayliteral && t.ty == Tarray && t.nextOf().ty == Tvoid && (cast(ArrayLiteralExp)e).elements.dim == 0);
 }
 
-// used by deduceType()
```
Comment (JinShil) [inline src/ddmd/dcast.d]: Seems like a useful comment.  I don't recommend deleting it.

**Review by JinShil — CHANGES_REQUESTED** [review]

File: src/ddmd/dcast.d (hunk 1 above)
Comment (RazvanN7) [reply src/ddmd/dcast.d]: it is right above the deduceType helper which is right above deduceType. A rgrep after rawTypeMerge will show that deduceType is the only function using it. The only way to get to that function is by reading the deduceType function. It made sense to have that comment when it was in a different file, but now it seems kind of useless.

Comment (JinShil) [reply src/ddmd/dcast.d]: Ok, I get it.  Thanks.

**Review by JinShil — APPROVED** [review]

**Review by PetarKirov — APPROVED** [review]

File: src/ddmd/dtemplate.d [hunk 2]
```diff
@@ -2918,6 +2919,27 @@ private ubyte deduceWildHelper(Type t, Type* at, Type tparam)
     }
 }
 
+private Type rawTypeMerge(Type t1, Type t2)
```
Comment (mathias-lang-sociomantic) [inline src/ddmd/dtemplate.d]: Documentation would be more than welcome

Comment (RazvanN7) [reply src/ddmd/dtemplate.d]: Done

**Pull request merged**